tests/test_linter/test_one_per_class.py

```diff
@@ -0,0 +1,161 @@
+    """
+    schema_view = SchemaView(test_schema)
+    config = RuleConfig(level=RuleLevel.error.text)
+
+    rule = OneIdentifierPerClass(config) if check_identifier else OneKeyPerClass(config)
+    return list(rule.check(schema_view, fix=False))
+
+
+@pytest.fixture(scope="module")
+def identifier_key_schema() -> str:
+    """Return a schema that defines a set of slots with identifiers and keys, and a test class to populate."""
+    return """
+# yaml-language-server: $schema=https://linkml.io/linkml-model/linkml_model/jsonschema/meta.schema.json
+id: https://w3id.org/linkml/examples/identifiers
+title: Identifiers, Keys, Primary Keys, and More!
+name: identifiers
+description: Schema for testing identifiers, keys, primary keys, and much, much more.
+
+prefixes:
+  ex: https://w3id.org/linkml/examples/patterns/
+
+default_prefix: ex
+
+slots:
+  slot_a:
+  slot_b:
+  id_slot_a:
+    identifier: true
+  id_slot_b:
+    identifier: true
+  id_slot_false:
+    identifier: false
+  key_slot_a:
+    key: true
+  key_slot_b:
+    key: true
+  key_slot_false:
+    key: false
+  pk_slot_a:
+    annotations:
+      primary_key: true
+  pk_slot_b:
+    annotations:
+      primary_key: true
+  pk_slot_false:
+    annotations:
+      primary_key: false
+
+classes:
+  TestClass:
+    slots:
+      - slot_a
+      - slot_b
+"""
+
+
+@pytest.mark.parametrize("id_slot_a", [EMPTY, "id_slot_a"])
+@pytest.mark.parametrize("id_slot_b", [EMPTY, "id_slot_b"])
+@pytest.mark.parametrize("id_slot_false", [EMPTY, "id_slot_false"])
+@pytest.mark.parametrize("key_slot_a", [EMPTY, "key_slot_a"])
+@pytest.mark.parametrize("key_slot_b", [EMPTY, "key_slot_b"])
+@pytest.mark.parametrize("key_slot_false", [EMPTY, "key_slot_false"])
+def test_get_identifier_get_key_slot(
+    identifier_key_schema: str,
+    id_slot_a: str,
+    id_slot_b: str,
+    id_slot_false: str,
+    key_slot_a: str,
+    key_slot_b: str,
+    key_slot_false: str,
+) -> None:
+    """Test the detection of multiple "identifier" and "key" slots in classes in the schema.
+
+    The parameters represent the slots to be added to the test class; if the value of the parameter is EMPTY,
+    the slot is not added. If the parameter is a string, that slot is added to the test class.
+
+    For example, given the following set of parameters:
+
+    id_slot_a: "id_slot_a"
+    id_slot_b: EMPTY
+    id_slot_false: "id_slot_false"
+    key_slot_a: EMPTY
+    key_slot_b: EMPTY
+    key_slot_false: "key_slot_false"
+
+    The following lines would be added to the bottom of the identifier_key_schema:
+
+          - id_slot_a
+          - id_slot_false
+          - key_slot_false
+
+    resulting in the following structure for `TestClass`:
+
+    classes:
+      TestClass:
+        slots:
+          - slot_a
+          - slot_b
+          - id_slot_a
+          - id_slot_false
+          - key_slot_false
+
+    This allows the testing of a large number of combinations without having to generate them all manually.
```

Comment:
To help grasp the pytest trick you could be explicit: "This and the staked parametrize decorators allow testing of all combinations (2**6=64) ..."